Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -804,6 +804,68 @@ TEST_F(SerializerTest, EncodesNonEmptyDocument) {
   ExpectRoundTrip(key, fields, update_time, proto);
 }
 
+TEST_F(SerializerTest,
+       BadBatchGetDocumentsResponseTagWithOtherValidTagsPresent) {
+  // A bad tag should be ignored, in which case, we should successfully
+  // deserialize the rest of the bytes as if it wasn't there. To craft these
+  // bytes, we'll use the same technique as
+  // EncodesFieldValuesWithRepeatedEntries (so go read the comments there
+  // first).
+
+  // Copy of the real one (from the nanopb generated firestore.pb.h), but with
+  // only missing and an extra_field
+  struct google_firestore_v1beta1_BatchGetDocumentsResponse_Fake {
+    pb_callback_t missing;
```

Comment:
Optional: perhaps clarify that `missing` is part of the normal response which means the document is missing? Upon first reading, I thought that this field was "missing" in the sense that it will be unset, and only then remembered that it belongs to `BatchGetDocumentsResponse`.

Reply:
Done.